### ISSUE TITLE
ws.upgradeReq undefined?

### ISSUE THREAD

**Issue by gianksp** [alexhultman/uWebSockets#76]
Awesome library! having a little problem finding the request parameters

```
//Sample call ws://config.app.ws_port?key=myapikey
var wss = new WebSocketServer({ httpServer:server, port:config.app.ws_port });
wss.on('connection', function(ws) {
  //Obtain the key on connection and validate it
  // var params = url.parse(ws.upgradeReq.url, true).query;
  // var key    = params.key;      
  console.log(ws.upgradeReq); //<-------- is undefined!!!!!?
});
```
am I missing something?

Thanks in advance

**Comment by alexhultman** [comment]
You cannot define a server and a port. If you define a server the port is set when calling listen on that server.

**Comment by alexhultman** [comment]
#64

**Comment by alexhultman** [comment]
@gianksp What if I made the upgradeReq only exist during the connection event? This object is humongous in memory size and completely destroys the very purpose of this project. They should have passed it as an argument so the user could decide if they wanted to save it in whole, save parts of it or forget about it.

**Comment by gianksp** [comment]
That would be great, I do think it is only necessary during connection. It would be a great addition and  the benefits outweighs the overhead during the connection.

**Comment by alexhultman** [comment]
Problem is if you have any kind of async thing happening at connection, the handler needs to explicitly capture the upgradeReq, not only the socket.

**Comment by alexhultman** [comment]
This issue will be solved in #97
